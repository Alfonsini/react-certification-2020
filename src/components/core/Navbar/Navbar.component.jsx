import React, { useState, useContext } from 'react';

import { useTheme } from 'styled-components';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars, faUser, faMoon, faSun } from '@fortawesome/free-solid-svg-icons';

import MenuHorizontal from '../HorizontalMenu/HorizontalMenu.component';

import {
  StyledMenu,
  StyledNav,
  StyledToggleThemeIcon,
  StyledToggleMenuIcon,
  StyledUlLeft,
  StyledUlRight,
  StyledLi,
  StyledToggleUserMenu,
} from './Navbar.styles';

import SearchForm from '../SearchForm';
import { ThemeContext } from '../../../providers/Theme/themeContext';
import { SKINS } from '../../../utils/constants';

function Navbar() {
  const [showMenu, setShowMenu] = useState(false);
  const { skin, setValue } = useContext(ThemeContext);
  const theme = useTheme();

  const toggleShowMenu = () => {
    setShowMenu(!showMenu);
  };

  const toggleSkin = () => {
    setValue(skin === SKINS.Light ? SKINS.Dark : SKINS.Light);
  };

  return (
    <>
      <StyledNav data-testid="nav">
        <StyledToggleMenuIcon>
          <FontAwesomeIcon
            data-testid="toggle-button"
            icon={faBars}
            onClick={toggleShowMenu}
            color={theme.iconColor}
            height="21px"
          />
        </StyledToggleMenuIcon>
        <StyledUlLeft>
          <StyledLi>
            <SearchForm />
          </StyledLi>
        </StyledUlLeft>
        <StyledUlRight>
          <StyledLi>
            <StyledToggleThemeIcon>
              <FontAwesomeIcon
                data-testid="toggle-theme-button"
                icon={skin === SKINS.Dark ? faSun : faMoon}
                onClick={toggleSkin}
                color={theme.iconColor}
                height="21px"
              />
            </StyledToggleThemeIcon>
          </StyledLi>
          <StyledLi>
            <StyledToggleUserMenu className="disabledCursor">
              <FontAwesomeIcon
                data-testid="user-button"
                icon={faUser}
                color={theme.iconColor}
                height="21px"
              />
            </StyledToggleUserMenu>
          </StyledLi>
        </StyledUlRight>
      </StyledNav>

      <StyledMenu data-testid="nav-menu-horizontal" active={showMenu}>
        <MenuHorizontal />
      </StyledMenu>
    </>
  );
}

export default Navbar;
