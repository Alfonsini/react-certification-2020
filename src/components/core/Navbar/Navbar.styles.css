.navbar-container {
  font-size: 18px;
  margin: 10px 10px;
  position: fixed;
  border-radius: 0.428rem;
  display: flex;
  padding: 1px;
  align-items: center;
  max-height: 80px;
  min-height: 80px;

  width: calc(100% - 110px);
}

.navbar-responsive-container {
  display: none;
  font-size: 18px;
  margin-top: 80px;
  padding-bottom: 0px;
}

.navbar-responsive-container.active {
  display: block;
}

.navbar-nav {
  list-style-type: none;
  display: flex;
  margin-right: 30px;
  align-items: center;
  width: 50%;
}

.navbar-responsive-nav {
  list-style-type: none;
  display: none;
}

.navbar-responsive-nav li {
  text-align: center;
  margin: 15px auto;
}

.nav-links {
  text-decoration: none;
}

.navbar-logo {
  margin-right: 30px;
}

.navbar-user {
  list-style-type: none;
  display: flex;
  margin-right: 30px;
  align-items: flex-end;
  right: 0;
  position: absolute;
}

.navbar-user a {
  color: #020635;
}

.navbar-toggle {
  display: none;
}

@media screen and (max-width: 500px) {
  .navbar-container {
    margin: 10px 10px;
  }

  .navbar-responsive-nav {
    display: block;
    margin-right: 30px;
    flex-direction: row;
    justify-content: flex-end;
  }

  .navbar-responsive-nav li {
    margin: 0;
  }

  .nav-links {
    margin: 0;
  }

  .navbar-logo {
    display: none;
  }

  .navbar-toggle {
    display: block;
    margin-left: 30px;
  }

  .navbar-user {
    margin-right: -30px;
  }
}
