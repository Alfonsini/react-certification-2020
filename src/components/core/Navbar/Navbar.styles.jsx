import styled from 'styled-components';

const StyledToggleIcon = styled.span`
  display: none;
  cursor: pointer;

  @media screen and (max-width: 500px) {
    display: block;
    margin-left: 30px;
  }
`;

const StyledToggleUserMenu = styled.span`
  text-decoration: none;
  cursor: pointer;
`;

const StyledMenu = styled.menu`
  display: none;
  font-size: 18px;
  margin-top: 80px;
  padding: 0;

  ${(props) =>
    props.active &&
    `
    display: block;
  `};
`;

const StyledNav = styled.nav`
  font-size: 18px;
  margin: 10px 10px;
  position: fixed;
  border-radius: 0.428rem;
  display: flex;
  padding: 1px;
  align-items: center;
  max-height: 80px;
  min-height: 80px;

  width: calc(100% - 110px);

  @media screen and (max-width: 500px) {
    margin: 10px 10px;
  }
`;

const StyledUlLeft = styled.ul`
  list-style-type: none;
  display: flex;
  margin-right: 30px;
  align-items: center;
  width: 50%;
`;

const StyledUlRight = styled.ul`
  list-style-type: none;
  display: flex;
  margin-right: 30px;
  align-items: flex-end;
  right: 0;
  position: absolute;

  @media screen and (max-width: 500px) {
    margin-right: -30px;
  }
`;

const StyledLi = styled.li`
  margin: 0;
`;

export {
  StyledMenu,
  StyledNav,
  StyledToggleIcon,
  StyledUlLeft,
  StyledUlRight,
  StyledLi,
  StyledToggleUserMenu,
};
