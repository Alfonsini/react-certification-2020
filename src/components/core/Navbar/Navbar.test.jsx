import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Navbar from '.';

beforeEach(() => {
  render(<Navbar />);
});

afterEach(() => {});

describe('Navbar', () => {
  test('render correctly', () => {
    const navbar = screen.getByTestId('nav');
    expect(navbar).toMatchSnapshot();
  });

  test('toggle button when visible', () => {
    const button = screen.getByTestId('toggle-button');
    expect(button).toMatchSnapshot();
    expect(button).toHaveClass('fa-bars');
  });

  test('user menu button when visible', () => {
    const button = screen.getByTestId('user-button');
    expect(button).toMatchSnapshot();
    expect(button).toHaveClass('fa-user');
  });

  test('render correctly', () => {
    const menu = screen.getByRole('menu', { hidden: true });
    expect(menu).toMatchSnapshot();
  });

  test('menu when visible', () => {
    const button = screen.getByTestId('toggle-button');
    fireEvent.click(button);

    const menu = screen.getByRole('menu');
    expect(menu).toBeVisible();
    expect(menu).toMatchSnapshot();
  });
});
