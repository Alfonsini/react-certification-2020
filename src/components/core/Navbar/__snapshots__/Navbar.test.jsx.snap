// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar menu when visible 1`] = `
<menu
  class="sc-gKsewC gSvkzp"
  role="menu"
>
  <ul
    class="sc-bdfBwQ cJHRYr"
    role="menu"
  >
    <li
      class="sc-gsTCUz fXCZoy"
    >
      <a
        class="sc-dlfnbm gWYOlF disabledCursor"
        to="/"
      >
        <span
          class="sc-hKgILt fNqzIf"
        >
          Home
        </span>
      </a>
    </li>
    <li
      class="sc-gsTCUz fXCZoy"
    >
      <a
        class="sc-dlfnbm gWYOlF disabledCursor"
        to="/"
      >
        <span
          class="sc-hKgILt fNqzIf"
        >
          Favorites
        </span>
      </a>
    </li>
  </ul>
</menu>
`;

exports[`Navbar render correctly 1`] = `
<nav
  class="sc-iBPRYJ jXdOCI"
  data-testid="nav"
>
  <span
    class="sc-eCssSg jdsWoj"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-bars fa-w-14 "
      data-icon="bars"
      data-prefix="fas"
      data-testid="toggle-button"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
      />
    </svg>
  </span>
  <ul
    class="sc-fubCfw dmnYev"
  >
    <li
      class="sc-jrAGrp hnzIIL"
    >
      <div
        class="search"
      >
        <input
          class="search-box"
          name="search"
          placeholder="Search..."
          type="text"
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-search fa-w-16 search-icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
      </div>
    </li>
  </ul>
  <ul
    class="sc-pFZIQ jgWwjd"
  >
    <li
      class="sc-jrAGrp hnzIIL"
    >
      <span
        class="sc-jSgupP gnqZgo disabledCursor"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user fa-w-14 "
          data-icon="user"
          data-prefix="fas"
          data-testid="user-button"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
          />
        </svg>
      </span>
    </li>
  </ul>
</nav>
`;

exports[`Navbar toggle button when visible 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-bars fa-w-14 "
  data-icon="bars"
  data-prefix="fas"
  data-testid="toggle-button"
  focusable="false"
  role="img"
  viewBox="0 0 448 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
    fill="currentColor"
  />
</svg>
`;

exports[`Navbar user menu button when visible 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-user fa-w-14 "
  data-icon="user"
  data-prefix="fas"
  data-testid="user-button"
  focusable="false"
  role="img"
  viewBox="0 0 448 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
    fill="currentColor"
  />
</svg>
`;
